### DISCUSSION TITLE
How to get current context when use javaagent?

### DISCUSSION THREAD

**Discussion by imperio-wxm** [open-telemetry/opentelemetry-java-instrumentation#7952] [Q&A]
version: 1.23.0-alpha
```java
-javaagent:/resources/opentelemetry-javaagent-all.jar
```
- I want to get the basic information of the trace after the jdbc statement, and then do other business processing based on this information.

- How to get the `current.context`, `traceId`,`statement.spanId` in the jdbc statement, I get is always empty{}.
```java
 public void selectAll(OpenTelemetryConnection connection) {
      String sql = "SELECT * FROM employees";

      Context parentContext = Context.current();
      try {
          OpenTelemetryStatement stmt = (OpenTelemetryStatement)connection.createStatement();
          ResultSet rs = stmt.executeQuery(sql);
          while (rs.next()) {
              System.out.println(rs.getInt("id") + "\t" +
                  rs.getString("name") + "\t" +
                  rs.getDouble("capacity"));
          }
      } catch (Exception e) {
          System.out.println(e.getMessage());
      }
  }
```

![image](https://user-images.githubusercontent.com/8111349/222380702-7628b624-1a11-41dd-8739-4c6f89f97840.png)

**Comment by imperio-wxm** [comment]
Can anyone answer this question? Or is it being used incorrectly?

**Comment by laurit** [comment]
Span is started and ended inside `executeQuery` method, you can't directly observe the span form outside that method. From the code snipped you pasted it seems that you are using the manual instrumentation for jdbc (did you disable javaagent jdbc instrumentation?). You could modify the jdbc library instrumentation to report what you need or you could build your own jdbc wrapper that you place between the opentelemetry instrumentation and the actual dirver where you could observe the active span. With javaagent instrumentation you could exploit the fact that it instruments all classes that implement `Statement` so you could just wrap the statement you have and the span would be create at the start of your wrapper method. See the source of the jdbc instrumentation https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/jdbc
